src/pydvl/utils/caching/disk.py

```diff
@@ -0,0 +1,116 @@
+import os
+import shutil
+from pathlib import Path
+from typing import Any, Optional, Union
+
+import cloudpickle
+
+from pydvl.utils.caching.base import CacheBackend
+
+__all__ = ["DiskCacheBackend"]
+
+PICKLE_VERSION = 5  # python >= 3.8
+
+DEFAULT_CACHE_DIR = Path().home() / ".pydvl_cache/disk"
+
+
+class DiskCacheBackend(CacheBackend):
+    """Disk cache backend that stores results in files.
+
+    Implements the CacheBackend interface for a disk-based cache.
+    Stores cache entries as pickled files on disk, keyed by cache key.
+    This allows sharing evaluations across processes in a single node/computer.
+
+    Args:
+        cache_dir: Base directory for cache storage.
+
+    Attributes:
+        cache_dir: Base directory for cache storage.
+
+    ??? Examples
+        ``` pycon
+        >>> from pydvl.utils.caching.disk import DiskCacheBackend
+        >>> cache_backend = DiskCacheBackend(cache_dir="/tmp/pydvl_disk_cache")
+        >>> cache_backend.clear()
+        >>> value = 42
+        >>> cache_backend.set("key", value)
+        >>> cache_backend.get("key")
+        42
+        ```
+
+        ``` pycon
+        >>> from pydvl.utils.caching.memcached import MemcachedCacheBackend
+        >>> cache_backend = MemcachedCacheBackend()
+        >>> cache_backend.clear()
+        >>> value = 42
+        >>> def foo(x: int):
+        ...     return x + 1
+        ...
+        >>> wrapped_foo = cache_backend.wrap(foo)
+        >>> wrapped_foo(value)
+        43
+        >>> wrapped_foo.stats.misses
+        1
+        >>> wrapped_foo.stats.hits
+        0
+        >>> wrapped_foo(value)
+        43
+        >>> wrapped_foo.stats.misses
+        1
+        >>> wrapped_foo.stats.hits
+        1
+        ```
+
+    """
+
+    def __init__(
+        self,
+        cache_dir: Union[os.PathLike, str] = DEFAULT_CACHE_DIR,
+    ) -> None:
+        """Initialize the disk cache backend.
+
+        Args:
+            cache_dir: Base directory for cache storage.
+                By default, this is set to `~/.pydvl_cache/disk`
+        """
+        super().__init__()
+        self.cache_dir = Path(cache_dir)
+        self.cache_dir.mkdir(exist_ok=True, parents=True)
```

Comment:
Wouln't it be better (and more portable) to use [tempfile](https://docs.python.org/3/library/tempfile.html)?